### PR TITLE
Improve documentation and update analysis.json

### LINKED ISSUE #42: Label, value and vaadin-item content.
Almost all of the participants find it very difficult to understand the correlations between the `label`, `value` and content of the `vaadin-item`.

screen shot 2017-11-10 at 10 28 50

It causes the problems with understanding of the `placeholder` displaying. 
One of the suggestions could be to introduce more descriptive documentation for using combinations of `label`, `value`, `placeholder` and content on `vaadin-item`.

For example, what will be shown when only `label`s are present, empty `value` + `placeholder` etc.

### LINKED ISSUE #43: Missing documentation for slots that can be used.
There is no notes and mentions about the `slot`s that can be used due to the fact we are using `vaadin-dropdown-menu-text-field` (wrapper for `vaadin-text-field`).

### PR DESCRIPTION
**Pull request by YuriyVaadin**
Fixes #42 
Fixes #43  

Improve documentation to include **hints** and user-specific cases about `label`s, `value`, `placeholder` and `slots`.

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-dropdown-menu/46)

### CODE REVIEW

**Comment by manolo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-dropdown-menu.html, line 180 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46#-KypONpoqsBzB13rDnVv:-KypONpoqsBzB13rDnVw:b-fhbcfu) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/cdcbd5b1c820b44d3ab8b56da7e75691f3b7a8e2/vaadin-dropdown-menu.html#L180)):*
> ```HTML
>        *
>        * Hint: You are able to change the visual representation of selected value in the input part
>        * of the dropdown-menu. To do so please set the `label`s to the vaadin-dropdown-menu options.
> ```

It's difficult to read and understand in a first view. I would not use options but vaadin-item instead, options in this sounds like parameters for the dropdown-menu.
I would rephrase as:

By setting the `label` property of inner `vaadin-item`s  you will be able to change the visual representation of the selected value in the input part.

---

*[vaadin-dropdown-menu.html, line 237 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46#-KypP7VapHLRAqAw6V5F:-KypP7VapHLRAqAw6V5G:bw6qwhn) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/cdcbd5b1c820b44d3ab8b56da7e75691f3b7a8e2/vaadin-dropdown-menu.html#L237)):*
> ```HTML
>              * Hint: not to select item by default set the value the way it won't match any
>              * dropdown-menu option's values or set all values of the options so they won't
>              * match the value of dropdown-menu itself.
> ```

Also hard to understand, rephrase to something like :

If you do not want to select any item by default, you can either set all the values of inner `vaadin-item`s, or set the `vaadin-dropdown-menu` value to an inexistent value in the items list.

---

*[vaadin-dropdown-menu.html, line 274 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46#-KypQDbOSMVgboH4tZhS:-KypQDbOSMVgboH4tZhT:bbr2njv) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/cdcbd5b1c820b44d3ab8b56da7e75691f3b7a8e2/vaadin-dropdown-menu.html#L274)):*
> ```HTML
>             /**
>              * A hint to the user of what can be entered in the control.
>              * Placeholder will be displayed only in case no items will be selected by default.
> ```

The placeholder will be displayed in the case that there is no item selected, or the selected item has an empty string label, or the selected item has no label and it's DOM content is empty.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46)*

**Comment by YuriyVaadin** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-dropdown-menu.html, line 180 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46#-KypONpoqsBzB13rDnVv:-KypSavIoXZ3xc8l07rD:by8plk2) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/cdcbd5b1c820b44d3ab8b56da7e75691f3b7a8e2/vaadin-dropdown-menu.html#L180)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

It's difficult to read and understand in a first view. I would not use options but vaadin-item instead, options in this sounds like parameters for the dropdown-menu.
I would rephrase as:

By setting the `label` property of inner `vaadin-item`s  you will be able to change the visual representation of the selected value in the input part.

</blockquote></details>

Thank for advices! Changed.

---

*[vaadin-dropdown-menu.html, line 237 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46#-KypP7VapHLRAqAw6V5F:-KypSe75N2XVCG9O4NE4:by8plk2) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/cdcbd5b1c820b44d3ab8b56da7e75691f3b7a8e2/vaadin-dropdown-menu.html#L237)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Also hard to understand, rephrase to something like :

If you do not want to select any item by default, you can either set all the values of inner `vaadin-item`s, or set the `vaadin-dropdown-menu` value to an inexistent value in the items list.
</blockquote></details>

Thank for advices! Changed.

---

*[vaadin-dropdown-menu.html, line 274 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46#-KypQDbOSMVgboH4tZhS:-KypSelk00DNcb6Fzz9w:by8plk2) ([raw file](https://github.com/vaadin/vaadin-dropdown-menu/blob/cdcbd5b1c820b44d3ab8b56da7e75691f3b7a8e2/vaadin-dropdown-menu.html#L274)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

The placeholder will be displayed in the case that there is no item selected, or the selected item has an empty string label, or the selected item has no label and it's DOM content is empty.
</blockquote></details>

Thank for advices! Changed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46)*

**Comment by manolo** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-dropdown-menu/46#-:-Kyph5-MgmBX-RVv6SN6:bnfp4nl)*
